src/common/colorspaces_inline_conversions.h

```diff
@@ -421,6 +422,125 @@ static inline float dt_camera_rgb_luminance(const float *const rgb)
   return (rgb[0] * 0.2225045f + rgb[1] * 0.7168786f + rgb[2] * 0.0606169f);
 }
 
+#ifdef _OPENMP
+#pragma omp declare simd
+#endif
+static inline void dt_XYZ_2_JzAzBz(const float *const XYZ_D65, float *const JzAzBz)
+{
+  static const float b = 1.15f;
+  static const float g = 0.66f;
+  static const float c1 = 0.8359375f; // 3424 / 2^12
+  static const float c2 = 18.8515625f; // 2413 / 2^7
+  static const float c3 = 18.6875f; // 2392 / 2^7
+  static const float n = 0.159301758f; // 2610 / 2^14
+  static const float p = 134.034375f; // 1.7 x 2523 / 2^5
+  static const float d = -0.56f;
+  static const float d0 = 1.6295499532821566e-11f;
+  static const float M[3][4] DT_ALIGNED_ARRAY = {
+      { 0.41478972f, 0.579999f, 0.0146480f, 0.0f },
+      { -0.2015100f, 1.120649f, 0.0531008f, 0.0f },
+      { -0.0166008f, 0.264800f, 0.6684799f, 0.0f },
+  };
+  static const float A[3][4] DT_ALIGNED_ARRAY = {
+      { 0.5f,       0.5f,       0.0f,      0.0f },
+      { 3.524000f, -4.066708f,  0.542708f, 0.0f },
+      { 0.199076f,  1.096799f, -1.295875f, 0.0f },
+  };
+
+  float XYZ[4] DT_ALIGNED_PIXEL = { 0.0f, 0.0f, 0.0f, 0.0f };
+  float LMS[4] DT_ALIGNED_PIXEL = { 0.0f, 0.0f, 0.0f, 0.0f };
+
+  // XYZ -> X'Y'Z
+  XYZ[0] = b * XYZ_D65[0] - (b - 1.0f) * XYZ_D65[2];
+  XYZ[1] = g * XYZ_D65[1] - (g - 1.0f) * XYZ_D65[0];
+  XYZ[2] = XYZ_D65[2];
+
+  // X'Y'Z -> L'M'S'
+  for(int i = 0; i < 3; i++)
+  {
+    LMS[i] = M[i][0] * XYZ[0] + M[i][1] * XYZ[1] + M[i][2] * XYZ[2];
+    LMS[i] = powf(fmax(LMS[i] / 10000.f, 0.0f), n);
+    LMS[i] = powf((c1 + c2 * LMS[i]) / (1.0f + c3 * LMS[i]), p);
+  }
+
+  // L'M'S' -> Izazbz
+  for(int i = 0; i < 3; i++) JzAzBz[i] = A[i][0] * LMS[0] + A[i][1] * LMS[1] + A[i][2] * LMS[2];
+  // Iz -> Jz
+  JzAzBz[0] = ((1.0f + d) * JzAzBz[0]) / (1.0f + d * JzAzBz[0]) - d0;
+}
+
+#ifdef _OPENMP
+#pragma omp declare simd
+#endif
+static inline void dt_JzAzBz_2_JzCzhz(const float *const JzAzBz, float *const JzCzhz)
+{
+  float var_H = atan2f(JzAzBz[2], JzAzBz[1]) / (2.0f * DT_M_PI_F);
+  JzCzhz[0] = JzAzBz[0];
+  JzCzhz[1] = sqrtf(JzAzBz[1] * JzAzBz[1] + JzAzBz[2] * JzAzBz[2]);
+  JzCzhz[2] = var_H >= 0.0f ? var_H : 1.0f + var_H;
+}
+
+#ifdef _OPENMP
+#pragma omp declare simd
+#endif
+static inline void dt_JzCzhz_2_JzAzBz(const float *const JzCzhz, float *const JzAzBz)
+{
+  JzAzBz[0] = JzCzhz[0];
+  JzAzBz[1] = cosf(2.0f * DT_M_PI_F * JzCzhz[2]) * JzCzhz[1];
```

Comment:
maybe store `2.0f * DT_M_PI_F` into a `#define DT_2_PI_F 2.0f * DT_M_PI_F` because I'm not sure the compiler is clever enough to premultiply the constant ahead instead of doing it at each call.

Reply:
#define won't change anything for compiler, because it's an instruction for preprocessor, which runs before the compiler. If DT_M_PI_F is a define too, then I'm very sure compiler will pre-calculate 2.0f * DT_M_PI_F. If it's a variable, then I guess what you need is another const variable.

Reply:
ok then.